From f4644b55d36a549710ba35b6fb797ba744807da6 Mon Sep 17 00:00:00 2001
From: Dmitry Dementyev <dementyev@google.com>
Date: Wed, 4 Oct 2023 15:02:03 -0700
Subject: [PATCH] Stop using deprecated getParcelable method in
 AccountManagerService.

Test: N/A
Bug: 299930871
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:7a24a283a360e816d2bfb2aa124c4eb2efd1be61)
Merged-In: I5a55acffe2e9ef2842c383a352b0bac66c6f1a55
Change-Id: I5a55acffe2e9ef2842c383a352b0bac66c6f1a55
---
 .../java/com/android/server/accounts/AccountManagerService.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/accounts/AccountManagerService.java b/services/core/java/com/android/server/accounts/AccountManagerService.java
index 7a19d034c2c8..5238595fe2a2 100644
--- a/services/core/java/com/android/server/accounts/AccountManagerService.java
+++ b/services/core/java/com/android/server/accounts/AccountManagerService.java
@@ -4923,7 +4923,7 @@ public class AccountManagerService
             p.setDataPosition(0);
             Bundle simulateBundle = p.readBundle();
             p.recycle();
-            Intent intent = bundle.getParcelable(AccountManager.KEY_INTENT);
+            Intent intent = bundle.getParcelable(AccountManager.KEY_INTENT, Intent.class);
             if (intent != null && intent.getClass() != Intent.class) {
                 return false;
             }
-- 
2.42.0.820.g83a721a137-goog

